src/Microsoft.Azure.WebJobs.Host/Bindings/ConverterManager.cs

```diff
@@ -331,273 +297,144 @@ public Type[] GetPossibleSourceTypesFromDestination(Type typeAttribute, Type typ
-                TDest obj = fromString(json, attr, context);
-                return obj;
-            };
+            return null;          
         }
 
-        // List of open converters. Since these are not exact type matches, need to search through and determine the match. 
-        private class Entry
-        {
-            public OpenType Source { get; set; }
-            public OpenType Dest { get; set; }
-            public Type Attribute { get; set; }
-
-            public Func<Type, Type, Func<object, object>> Builder { get; set; }
-        }
-
-        // "Empty" converter manager that only allows identity conversions. 
-        // This is useful for constrained rules that don't want to operate against exact types and skip 
-        // arbitrary user conversions. 
-        private class IdentityConverterManager : IConverterManager
+        // Compose via a middle type. 
+        // If we have (TSrc --> X) and (X --> TDest), then support (Tsrc-->TDest) via X
+        private FuncAsyncConverter GetMiddle<TAttribute, TMiddle>(Type typeSource, Type typeDest)
+            where TAttribute : Attribute
         {
-            public void AddConverter<TSource, TDestination, TAttribute1>(FuncConverter<TSource, TAttribute1, TDestination> converter) where TAttribute1 : Attribute
+            var first = TryGetConverter<TAttribute>(typeSource, typeof(TMiddle));
+            if (first != null)
             {
-                throw new NotImplementedException();
-            }
-
-            public void AddConverter<TSrc, TDest, TAttribute1>(Func<Type, Type, Func<object, object>> converterBuilder) where TAttribute1 : Attribute
-            {
-                throw new NotImplementedException();
-            }
-
-            public FuncConverter<TSource, TAttribute1, TDestination> GetConverter<TSource, TDestination, TAttribute1>() where TAttribute1 : Attribute
-            {
-                if (typeof(TSource) != typeof(TDestination))
+                var second = TryGetConverter<TAttribute>(typeof(TMiddle), typeDest);
+                if (second != null)
                 {
-                    return null;
+                    // TSrc --> Jobject --> TDest
+                    return async (src, attr, context) =>
+                    {
+                        TMiddle jobj = (TMiddle)await first(src, attr, context);
+                        object obj = await second(jobj, attr, context);
+                        return obj;
+                    };
                 }
-                return (src, attr, ctx) =>
-                {
-                    object obj = (object)src;
-                    return (TDestination)obj;
-                };
             }
+            return null;
         }
 
-        // Match a generic type with 1 generic arg. 
-        // like IEnumerable<T>,  IQueryable<T>, etc. 
-        private class SingleGenericArgOpenType : OpenType
+
+        // List of all converters. This may refer to an pen type or an exact match. 
```

Comment:
This comment refers to a list, but is on a class? Also, can you use a better name for the class? What does it represent?